drivers/timer/Kconfig.native_posix

```diff
@@ -6,7 +6,7 @@
 config NATIVE_POSIX_TIMER
 	bool "(POSIX) native_posix timer driver"
 	default y
```

Comment:
Then, please set this default to y only if BOARD_NATIVE_POSIX, as there is other out of tree boards based on the POSIX architecture.

Reply:
It's possible to just remove the `default y` here, since it's already in `boards/posix/native_posix/Kconfig` via `select NATIVE_POSIX_TIMER`.

Additionally, `nrf52_bsim` has a `select NRF_RTC_TIMER`.

It would probably make sense to keep the `depends ARCH_POSIX` though.

Reply:
> It would probably make sense to keep the depends ARCH_POSIX though.

It would. But it also does make sense with the depends on BOARD_NATIVE_POSIX , as it does use that board specific code :). If you really want to go this way, maybe better add a comment here, saying that this driver does require the native_posix board HW models.

Reply:
Would it make more sense to create a Device tree binding and use `DT_HAS_NATIVE_POSIX_TIMER_ENABLED`?

With that, it would be a bit more explicit that the board has a hardware model that is compatible with `native_posix`.

Reply:
No. My main concern with this change is that there is no real HW (or SW component) you can claim to also have in another target board and therefore reuse the driver. There is just internal APIs of the native_posix source.